shell/common/animator.h

```diff
@@ -87,6 +91,8 @@ class Animator final {
 
   fml::WeakPtrFactory<Animator> weak_factory_;
 
+  friend class testing::ShellTest_FrameRasterizedCallbackIsCalled_Test;
```

Comment:
Look at the `FRIEND_TEST` macro in `gtest_prod.h` that add this. If you have trouble importing that header due to the `gn check`, you can append `//nocheck` to the header include. The production header is meant exactly for uses like this.

Reply:
The complication mainly comes from the `namespace testing` inside `namespace flutter`. I can use the `FRIEND_TEST` but I still have to forward declare the test class manually. Can we drop the `namespace testing` (maybe in a later PR)?

Reply:
Nevermind, I managed to encapsulate the call inside `ShellTest` fixture (for reuse) so I only need to friend the `ShellTest`.